### ISSUE TITLE
[1.3.0-RC1] sbt.librarymanagement.ResolveException: Error downloading org.webjars.npm:dom-serializer:[0,1)

### ISSUE THREAD

**Issue by gavinbaumanis** [sbt/sbt#4708]
- [ ] I've read the [CONTRIBUTING](https://github.com/sbt/sbt/blob/develop/CONTRIBUTING.md) guidelines

## steps
Running any SBT command 

## problem
All commands fail with the same stack trace produced

## expectation
SBT commands to complete successfully.

## notes

sbt version: 1.3.0-RC1
Scala 1.2.8

```
[warn]  Note: Unresolved dependencies path:
[error] sbt.librarymanagement.ResolveException: Error downloading org.webjars.npm:dom-serializer:[0,1)
[error]   Not found
[error]   not found: https://repo1.maven.org/maven2/org/webjars/npm/dom-serializer/%5B0,1)/dom-serializer-%5B0,1).pom
[error]   not found: C:\Users\Gavin Baumanis\.ivy2\local\org.webjars.npm\dom-serializer\[0,1)\ivys\ivy.xml
[error]   download error: Caught java.net.URISyntaxException: Illegal character in path at index 87: https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.webjars.npm/dom-serializer/[0,1)/ivys/ivy.xml (Illegal character in path at index 87: https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.webjars.npm/dom-serializer/[0,1)/ivys/ivy.xml) while downloading https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.webjars.npm/dom-serializer/[0,1)/ivys/ivy.xml
[error]         at lmcoursier.CoursierDependencyResolution.unresolvedWarningOrThrow(CoursierDependencyResolution.scala:227)
[error]         at lmcoursier.CoursierDependencyResolution.$anonfun$update$30(CoursierDependencyResolution.scala:196)
[error]         at scala.util.Either$LeftProjection.map(Either.scala:573)
[error]         at lmcoursier.CoursierDependencyResolution.update(CoursierDependencyResolution.scala:196)
[error]         at sbt.librarymanagement.DependencyResolution.update(DependencyResolution.scala:56)
[error]         at sbt.internal.LibraryManagement$.resolve$1(LibraryManagement.scala:87)
[error]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$12(LibraryManagement.scala:137)
[error]         at sbt.util.Tracked$.$anonfun$lastOutput$1(Tracked.scala:69)
[error]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$20(LibraryManagement.scala:150)
[error]         at scala.util.control.Exception$Catch.apply(Exception.scala:228)
[error]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11(LibraryManagement.scala:150)
[error]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11$adapted(LibraryManagement.scala:131)
[error]         at sbt.util.Tracked$.$anonfun$inputChanged$1(Tracked.scala:150)
[error]         at sbt.internal.LibraryManagement$.cachedUpdate(LibraryManagement.scala:164)
[error]         at sbt.Classpaths$.$anonfun$updateTask0$5(Defaults.scala:2844)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:67)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:270)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
[error]         at sbt.Execute.work(Execute.scala:279)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:270)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error]         at java.base/java.lang.Thread.run(Thread.java:835)
[error] (update) sbt.librarymanagement.ResolveException: Error downloading org.webjars.npm:dom-serializer:[0,1)
[error]   Not found
[error]   not found: https://repo1.maven.org/maven2/org/webjars/npm/dom-serializer/%5B0,1)/dom-serializer-%5B0,1).pom
[error]   not found: C:\Users\Gavin Baumanis\.ivy2\local\org.webjars.npm\dom-serializer\[0,1)\ivys\ivy.xml
[error]   download error: Caught java.net.URISyntaxException: Illegal character in path at index 87: https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.webjars.npm/dom-serializer/[0,1)/ivys/ivy.xml (Illegal character in path at index 87: https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.webjars.npm/dom-serializer/[0,1)/ivys/ivy.xml) while downloading https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.webjars.npm/dom-serializer/[0,1)/ivys/ivy.xml
```

**Comment by eed3si9n** [comment]
@gavinbaumanis Thanks for the report. It seems like this happens with specific library or a plugin, do you have a project that we can use on our computer to reproduce the issue?

**Comment by gavinbaumanis** [comment]
@eed3si9n  : I am just creating a new application from the play-scala-seed - using my build / plugins and configuration files.

I'll load up things one at a time, then two / etc... and see if I can find the offending item.
Update : It is going to take a while to go through all the scenarios and document them all properly.
I'll update the thread once I have completed testing / documenting.

**Comment by gavinbaumanis** [comment]
I wrote a BATCH script to;
* Add one line at a time from my build.sbt and plugins.sbt files into the test application..
* And to alternate between a build.properties of 1.2.8 and 1.3.0-RC1
* And to alternate between sbt executable versions of 1.2.8 and 1.3.0-RC1

And finally found the offender, It is a plugin; JSHINT.
`addSbtPlugin("com.typesafe.sbt" % "sbt-jshint" % "1.0.6")`

The issue is only evident (on my system), when 
* 1.3.0-RC1, specified in the build.properties
* using 1.2.8 or 1.3.0-RC1 as the sbt executable, both fail.

SCALA :
1.2.8

JAVA :
openjdk version "12.0.1" 2019-04-16
OpenJDK Runtime Environment (build 12.0.1+12)
OpenJDK 64-Bit Server VM (build 12.0.1+12, mixed mode, sharing)

OS :
WINDOWS 10 (Latest INSIDER Release)

I have attached a ZIP that will produce the error;
*  the play-scala-seed
* build.sbt
* plugins.sbt
* build.properties
[4708.zip](https://github.com/sbt/sbt/files/3197956/4708.zip)

**Comment by guersam** [comment]
Something similar happens in OSX with a local path containing a space:
https://gist.github.com/guersam/62378304a0cd89e7dda61412232a4410

**Comment by bitnot** [comment]
Maybe not related, but the same issue (`dom-serializer-%5B0,1).pom`) happens in one of my project when using `sbt-coursier` with SBT 1.2.8.
Added a sample project to reproduce: https://github.com/bitnot/sbt-coursier-dom-serializer-issue

**Comment by eed3si9n** [comment]
The workaround would be

```scala
ThisBuild / useCoursier := false
```

**Comment by bitnot** [comment]
SBT 1.3.0-RC1 and sbt-coursier fail to resolve `org.webjars.npm:entities`.

Here's how SBT 1.2.8 resolves the conflict:
```
+-com.typesafe.sbt:sbt-jshint:1.0.6
| +-org.webjars.npm:jshint:2.9.3
| | +-org.webjars.npm:htmlparser2:3.8.3
| | | +-org.webjars.npm:domutils:1.5.1
| | | | +-org.webjars.npm:dom-serializer:0.1.1
| | | | | +-org.webjars.npm:entities:1.0.0
| | | | | +-org.webjars.npm:entities:[1.1.1,2) (evicted by by: 1.0.0)
| | | +-org.webjars.npm:entities:1.0.0
...
```

Possible workaround: force `libraryDependencies += "org.webjars.npm" % "domutils" % "1.5.0"` in `project/plugins.sbt` to remove `dom-serializer` dependency.

**Comment by gavinbaumanis** [comment]
Just to confirm;
ThisBuild / useCoursier := false
Did not fix the issue 

Adding;
libraryDependencies += "org.webjars.npm" % "domutils" % "1.5.0" into project/plugins.sbt

did allow me to compile the application with the jshint plugin.

**Comment by ches** [comment]
I believe the `URISyntaxException` and Coursier dealing with lower bounds of 0 was resolved by #4845. 

I still see issues on Play apps with sbt-jshint where Coursier errors out with a conflicting dependency error for two versions of the `entities` webjar, like @bitnot reports above. It's a separate issue from the `[0` one though, possibly a symptom of coursier/coursier#569. I'll try to reproduce with latest sbt 1.3.0 and open a new issue detailing it, if we want to track it outside of Coursier (I've so far encountered it with sbt 1.2.8 and sbt-coursier 2.0.0-RC2, I'd expect the same issue under current lm-coursier).

**Comment by eed3si9n** [comment]
ok. I'll close this one.
